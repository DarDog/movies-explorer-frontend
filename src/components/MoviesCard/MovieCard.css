.movies__item {
  display: flex;
  justify-content: space-between;
  background-color: #F9F9F9;
  padding: 6px;
  height: 100%;
}

.movies__link {
  width: 100%;
  max-width: 220px;
}

.movies__item:last-of-type {
  margin-bottom: 0;
}

.movies__info {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-items: flex-start;
  padding: 16px;
}

.movies__title {
  margin: 0 0 13px;
  padding: 0;
  font-weight: 500;
  font-size: 18px;
  line-height: 22px;
  color: #000;
}

.movies__duration {
  margin: 0 0 30px;
  padding: 0;
  font-weight: 400;
  font-size: 11px;
  line-height: 13px;
  color: #A0A0A0;
}

.movies__like {
  border: none;
  margin: 0;
  background: url("data:image/svg+xml,%3Csvg width='14' height='12' viewBox='0 0 14 12' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M6.65242 1.89789L7.01419 2.24773L7.36168 1.8837C8.08219 1.12888 8.97817 0.5 10.1818 0.5C12.1019 0.5 13.5 2.02862 13.5 4C13.5 4.9368 13.0747 5.73587 12.3847 6.40496L7 11.3228L1.60992 6.40004L1.59988 6.39087L1.58936 6.38227C0.885614 5.80642 0.5 4.96765 0.5 4C0.5 2.02862 1.89813 0.5 3.81818 0.5C5.01333 0.5 5.90847 1.17846 6.65242 1.89789Z' stroke='%23E8E8E8'/%3E%3C/svg%3E%0A") no-repeat center;
  background-size: contain;
  padding: 6px 7px;
  transition: transform .3s;
  z-index: 100;
}

.movies__like_active {
  background: url("data:image/svg+xml,%3Csvg width='14' height='12' viewBox='0 0 14 12' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M10.1818 0C8.78182 0 7.76364 0.738462 7 1.53846C6.23636 0.8 5.21818 0 3.81818 0C1.59091 0 0 1.78462 0 4C0 5.10769 0.445455 6.09231 1.27273 6.76923L7 12L12.7273 6.76923C13.4909 6.03077 14 5.10769 14 4C14 1.78462 12.4091 0 10.1818 0Z' fill='%23EE3465'/%3E%3C/svg%3E%0A") no-repeat center;
}

.movies__like:hover {
  transform: scale(1.5);
}

.movies__poster {
  max-width: 220px;
  max-height: 122px;
  object-fit: cover;
  object-position: center;
  width: 100%;
  height: 100%;
  align-self: center;
}

.movies__delete {
  background: url("data:image/svg+xml,%3Csvg width='10' height='10' viewBox='0 0 10 10' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M5.00011 6.06054L8.00526 9.06569L9.06592 8.00503L6.06077 4.99988L9.06581 1.99484L8.00515 0.934182L5.00011 3.93922L1.99486 0.93396L0.934195 1.99462L3.93945 4.99988L0.934082 8.00525L1.99474 9.06591L5.00011 6.06054Z' fill='black'/%3E%3C/svg%3E%0A") no-repeat center;
  border: none;
  padding: 5px 5px;
  transition: transform .3s;
  z-index: 100;
}

.movies__delete:hover {
  transform: scale(1.5);
}

@media screen and (max-width: 500px){
  .movies__item {
    padding: 0;
    flex-direction: column-reverse;
  }

  .movies__link {
    width: 100%;
    max-width: 480px;
  }

  .movies__poster {
    max-width: 100%;
    max-height: 168px;
  }

  .movies__info {
    padding: 14px 12px;
    display: grid;
    grid-template-columns: 3fr 1fr;
    grid-template-rows: repeat(1, 1fr);
    gap: 8px;
  }

  .movies__title {
    grid-column: 1;
    grid-row: 1;
    margin: 0;
    font-size: 12px;
    line-height: 15px;
  }

  .movies__duration {
    grid-column: 1;
    grid-row: 2;
    margin: 0;
  }

  .movies__like {
    grid-row: 1/3;
    grid-column: 2;
    justify-self: end;
  }

  .movies__delete {
    grid-row: 1/3;
    grid-column: 2;
    justify-self: end;
  }
}
