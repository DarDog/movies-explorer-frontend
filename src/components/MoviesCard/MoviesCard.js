import React, { useState, useEffect } from 'react';
import './MovieCard.css';
import { mainApi } from "../../utils/MainApi";
import { useAuth } from "../../hook/useAuth";

const MoviesCard = ({ movie, ...props }) => {
  const [isSaved, setIsSaved] = useState(false);
  const savedMovies = JSON.parse(localStorage.getItem('saved-movies'));
  const { getSavedMovies } = useAuth();

  const handleLike = () => {
    if (!isSaved) {
      mainApi.setSavedMovie({
        country: movie.country,
        director: movie.director,
        duration: movie.duration,
        year: movie.year,
        description: movie.description,
        image: `https://api.nomoreparties.co${ movie.image.url }`,
        trailer: movie.trailerLink,
        thumbnail: `https://api.nomoreparties.co${ movie.image.formats.thumbnail.url }`,
        movieId: movie.id,
        nameRU: movie.nameRU,
        nameEN: movie.nameEN,
      })
        .then(() => {
          setIsSaved(true);
          getSavedMovies();
        })
        .catch(err => console.error(err))
    } else {
      handleRemove();
    }
  }

  const handleRemove = () => {
    mainApi.removeSavedMovie(movie.id)
      .then(() => {
        setIsSaved(false);
        getSavedMovies();
      })
      .catch(err => console.error(err))
  }

  useEffect(() => {
    const match = savedMovies.find(savedMovie => savedMovie.movieId === movie.id)

    if (match) {
      setIsSaved(true);
    }
  }, [])

  return (
    <li className='movies__item'>
      <div className="movies__info">
        <h2 className="movies__title">{ movie.nameRU }</h2>
        <p className="movies__duration">{ Math.floor(movie.duration / 60) }ч { movie.duration % 60 }м</p>
        { props.isSaves
          ? <button type='button' className={ `movies__delete` } onClick={handleRemove}/>
          : <button type='button' className={ `movies__like ${ isSaved && 'movies__like_active' }` } onClick={handleLike}/>
        }
      </div>
      <img src={ `https://api.nomoreparties.co/${ movie.image.url }` } alt={ movie.title } className="movies__poster"/>
    </li>
  );
}

export default MoviesCard;
