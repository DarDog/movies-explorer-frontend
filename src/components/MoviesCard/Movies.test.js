import React from 'react';
import MoviesCard from "./MoviesCard";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import movies from "../../utils/preparedFilms";

describe('Компонент MoviesCard', () => {
  it('Рендерется', () => {
    render(<MoviesCard  movie={movies[0]} />);

    expect(screen.getByRole('listitem')).toBeInTheDocument();
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(screen.getByRole('img')).toBeInTheDocument();
  });
  it('Соответствует Snapshot', () => {
    const component = render(
      <MoviesCard movie={movies}/>
    )

    expect(component).toMatchSnapshot();
  });
})
