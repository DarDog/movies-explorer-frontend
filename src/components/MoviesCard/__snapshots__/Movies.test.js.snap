// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент MoviesCard Соответствует Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="movies__item"
      >
        <div
          class="movies__info"
        >
          <h2
            class="movies__title"
          />
          <p
            class="movies__duration"
          >
            NaN
            ч 
            NaN
            м
          </p>
          <button
            class="movies__like undefined"
            type="button"
          />
        </div>
        <img
          alt="Заглушка"
          class="movies__poster"
        />
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="movies__item"
    >
      <div
        class="movies__info"
      >
        <h2
          class="movies__title"
        />
        <p
          class="movies__duration"
        >
          NaN
          ч 
          NaN
          м
        </p>
        <button
          class="movies__like undefined"
          type="button"
        />
      </div>
      <img
        alt="Заглушка"
        class="movies__poster"
      />
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
